runtime/moonbase/src/lib.rs

```diff
@@ -649,8 +649,9 @@ impl pallet_author_mapping::Config for Runtime {
 	type AuthorId = NimbusId;
 	type DepositCurrency = Balances;
 	type DepositAmount = DepositAmount;
-	fn can_register(account: &AccountId) -> bool {
-		ParachainStaking::is_candidate(account)
+	fn can_register(_account: &AccountId) -> bool {
+		// ParachainStaking::is_candidate(account)
+		true // to benchmark author mapping, which cannot register candidates in staking
 	}
```

Comment:
I can change this back, but it was necessary for weight generation to work.

Reply:
Moreover, weight generation understates the actual complexity because I made this change to make it work. For `add-association`, there is 1 storage read (to staking) which is not counted in the current implementation.

Reply:
Would whitelisting storage keys help here?

Reply:
Not sure, the problem is that `benchmarking.rs` in `author-mapping` does not have the context necessary to call `ParachainStaking::join_candidates`, but the benchmarking done at runtime requires that this is called successfully before an account registers an author mapping.

Reply:
Maybe this is part of why it is uncommon to see associated functions in pallet config traits. An alternative approach would be to make an associated type for this and benchmark it along with parachain staking. That pattern is more common in Parity's pallet.

But honestly, I think removing the check as you did in #600 makes sense.